code-conventions.md

```diff
@@ -75,16 +116,19 @@ code. Contributors should also see the dedicated
   - Effect: init ->  execute  -> toString
   - Condition: init -> check -> toString
 
-## Naming
-* Classes named in CapitalCamelCase
-* Fields and methods named in camelCase
-  - Final static fields may be named in SCREAMING_SNAKE_CASE
-* Localized messages should be named like m_message_name
+
+### Naming
+* Class names are written in `UpperCamelCase`
+  - The file name should match its primary class name (e.g. `MyClass` goes in `MyClass.java`.)
+* Fields and methods named in `camelCase`
+  - Static constant fields should be named in `UPPER_SNAKE_CASE`
+* Localised messages should be named in `lower_snake_case`
   - And that is the only place where snake_case is acceptable
-* Use prefixes where their use has been already estabilished (such as ExprSomeRandomThing)
-  - Otherwise, use postfixes (such as LoopSection)
+* Use prefixes only where their use has been already estabilished (such as `ExprSomeRandomThing`)
+  - Otherwise, use postfixes where necessary (such as `LoopSection`)
+  - Please refer to existing files for examples
```

Comment:
```suggestion
* Use prefixes only where their use has been already estabilished (such as `ExprSomeRandomThing`)
  - Otherwise, use postfixes where necessary
  - Common occurrences include: Struct (Structure), Sec (Section), EffSec (EffectSection), Eff (Effect), Cond (Condition), Expr (Expression)
```